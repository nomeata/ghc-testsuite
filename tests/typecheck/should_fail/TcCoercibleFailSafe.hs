{-# LANGUAGE RoleAnnotations, RankNTypes, ScopedTypeVariables, Safe #-}

import GHC.Prim (coerce, Coercible)
import Data.Ord (Down)

newtype Age = Age Int deriving Show

foo1 :: (Down Age -> Down Int)
foo1 = coerce 

main = return ()
