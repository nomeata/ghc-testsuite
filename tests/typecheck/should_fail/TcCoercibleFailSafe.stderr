
TcCoercibleFailSafe.hs:9:8:
    No instance for (Coercible (Down Age) (Down Int))
      because the constructor of ‛Down’ is not imported
      as required in SafeHaskell mode
      arising from a use of ‛coerce’
    In the expression: coerce
    In an equation for ‛foo1’: foo1 = coerce
